File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPEnderChest.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPEnderChest implements PlayerStateProperty<ItemStack[]> {
    @Override
    public String getConfigName() {
        return "ender_chest";
    }

    @Override
    public ItemStack[] getDefaultSerializedProperty() {
        return new ItemStack[0];
    }

    @Override
    public ItemStack[] getSerializedPropertyFromPlayer(Player player) {
        return player.getEnderChest().getContents();
    }

    @Override
    public void restoreProperty(Player player, ItemStack[] chestContents) {
        player.getEnderChest().setContents(chestContents);
    }

    @Override
    public Class<ItemStack[]> getTypeClass() {
        return ItemStack[].class;
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPEnderChest.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.noxet.noxetserver.playerstate.PlayerStateProperty

import scala.jdk.CollectionConverters.*

object PSPEnderChest extends PlayerStateProperty[Array[ItemStack]]:
  override def getConfigName: String = "ender_chest"

  override def getDefaultSerializedProperty: Array[ItemStack] = Array.empty

  override def getSerializedPropertyFromPlayer(player: Player): Array[ItemStack] =
    player.getEnderChest.getContents

  override def restoreProperty(player: Player, chestContents: Array[ItemStack]): Unit =
    player.getEnderChest.setContents(chestContents)

  override def getValueFromConfig(config: ConfigurationSection): Array[ItemStack] =
    config.getList(getConfigName).asScala.toArray